x/budget/types/utils_test.go

```diff
@@ -15,6 +15,55 @@ func TestParseTime(t *testing.T) {
 	require.NoError(t, err)
 	errorCase := "9999-12-31T00:00:00_ErrorCase"
 	_, err = time.Parse(time.RFC3339, errorCase)
-	require.PanicsWithError(t, err.Error(), func() { types.ParseTime(errorCase) })
-	require.Equal(t, normalRes, types.ParseTime(normalCase))
+	require.PanicsWithError(t, err.Error(), func() { types.MustParseRFC3339(errorCase) })
+	require.Equal(t, normalRes, types.MustParseRFC3339(normalCase))
+}
+
+func TestDateRageOverlap(t *testing.T) {
```

Comment:
Maybe we can add more tests for various cases. If you don't mind I can add these tests in this PR.

Reply:
Of course, That would be helpful, Thanks.